src/app/src/test/kotlin/org/imperial/mrc/hint/integration/DownloadTests.kt

```diff
@@ -0,0 +1,73 @@
+package org.imperial.mrc.hint.integration
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.EnumSource
+import org.springframework.boot.test.web.client.getForEntity
+import org.springframework.boot.test.web.client.postForEntity
+import org.springframework.http.HttpEntity
+import org.springframework.http.HttpHeaders
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+
+class DownloadTests : SecureIntegrationTests() {
+
+    private fun getModelRunJsonEntity(): HttpEntity<String> {
+        val options = mapOf("sleep" to 1)
+        val headers = HttpHeaders()
+        headers.contentType = MediaType.APPLICATION_JSON
+        val jsonString = ObjectMapper().writeValueAsString(options)
+        return HttpEntity(jsonString, headers)
+    }
+
+    private fun waitForModelRunResult(isAuthorized: IsAuthorized) : String {
+        val id = when (isAuthorized) {
+            IsAuthorized.TRUE -> {
+                val entity = getModelRunJsonEntity()
+                val runResult = testRestTemplate.postForEntity<String>("/model/run/", entity)
+                ObjectMapper().readValue<JsonNode>(runResult.body!!)["data"]["id"].textValue()
+            }
+            IsAuthorized.FALSE -> "nonsense"
+        }
+
+        do {
+            Thread.sleep(500)
+            val statusResponse = testRestTemplate.getForEntity<String>("/model/status/$id")
```

Comment:
I do wonder if this is going to re-introduce the non-deterministic test failure we just removed, since this is the same logic that was failing and I didn't actually get to the bottom of why, I just ditched it. Can we just stick some logging in here to get ahead of it? Just logging this statusResponse would be good.